src/mlx/warnings/regex_checker.py

```diff
@@ -107,12 +161,156 @@ def check(self, content):
+        'intentional': 'info',
+        'bug': 'major',
+        'unclassified': 'major',
+        'pending': 'critical',
+    }
+
+    def __init__(self, classification, **kwargs):
+        """Initialize the CoverityClassificationChecker:
+
+        Args:
+            classification (str): The coverity classification
+        """
+        super().__init__(**kwargs)
+        self.classification = classification
+
+    @property
+    def cq_description_template(self):
+        ''' Template: string.Template instance based on the configured template string '''
+        return self._cq_description_template
+
+    @cq_description_template.setter
+    def cq_description_template(self, template_obj):
+        self._cq_description_template = template_obj
+
+    def return_count(self):
+        ''' Getter function for the amount of warnings found
+
+        Returns:
+            int: Number of warnings found
+        '''
+        return self.count
+
+    def add_code_quality_finding(self, match):
+        '''Add code quality finding
+
+        Args:
+            match (re.Match): The regex match
+        '''
+        finding = {
+            "severity": "major",
+            "location": {
+                "path": self.cq_default_path,
+                "positions": {
+                    "begin": {
+                        "line": 1,
+                        "column": 1
+                    }
+                }
+            }
+        }
+        groups = {name: result for name, result in match.groupdict().items() if result}
+        try:
+            description = self.cq_description_template.substitute(os.environ, **groups)
+        except KeyError as err:
+            raise WarningsConfigError(f"Failed to find environment variable from configuration value "
+                                      f"'cq_description_template': {err}") from err
+        if "classification" in groups:
+            finding["severity"] = self.SEVERITY_MAP[groups.get("classification", "unclassified").lower()]
+        if "path" in groups:
+            path = Path(groups["path"])
+            if path.is_absolute():
+                try:
+                    path = path.relative_to(Path.cwd())
+                except ValueError as err:
+                    raise ValueError("Failed to convert abolute path to relative path for Code Quality report: "
+                                     f"{err}") from err
+            finding["location"]["path"] = str(path)
+        if "line" in groups:
+            try:
+                line_number = int(groups["line"], 0)
+            except (TypeError, ValueError):
+                line_number = 1
+            finding["location"]["positions"]["begin"]["line"] = line_number
+        if "col" in groups:
+            try:
+                column_number = int(groups["col"], 0)
+            except (TypeError, ValueError):
+                column_number = 1
+            finding["location"]["positions"]["begin"]["column"] = column_number
```

Comment:
```suggestion
                finding["location"]["positions"]["begin"]["column"] = int(groups["col"], 0)
            except (TypeError, ValueError):
                pass
```
The default value of `1` has already been stored. When can a `TypeError` or `ValueError` be raised?

You could get rid of duplicate code with `for group_name in ("line", "column"):` if you use `column` instead of `col` as group name.

Reply:
TypeError can be raised when it's for example a list. ValueError can be raised when it's for example a string that can not be converted to an integer.
I can add KeyError to catch the error when `"col"` does not exist as key.

Reply:
Resolved in d9f92e09d3214c72c8972fd9cb0c32adb501f60a